sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -560,9 +560,10 @@ public Node visitInsert(SqlBaseParser.InsertContext context) {
             }
             throw e;
         }
-        return new Insert(
+        Node subquery = visit(context.insertSource().query());
+        return new Insert<>(
             table,
-            (Query) visit(context.insertSource().query()),
+            subquery == null ? null : (Query) subquery,
```

Comment:
As far as I understand subquery can be null, right ?

Reply:
`null` would mean that the insert source is missing https://github.com/crate/crate/blob/master/sql-parser/src/main/antlr/SqlBase.g4#L441-L442 . I don't think that it is the case.

Also, do we want to keep calling it subquery? `insertSource().query()` can be either querySpec (subqquery) or values, therefore I am not sure that it is suitable name.